internal/npm_install/generate_build_file.js

```diff
@@ -155,6 +158,41 @@ function isPackage(p) {
       fs.statSync(packageJson).isFile();
 }
 
+/**
+ * Returns true if directory is a Bazel package directory.
+ */
+function isPackageDir(d) {
+  return fs.existsSync(path.posix.join(d, 'BUILD')) ||
+      fs.existsSync(path.posix.join(d, 'BUILD.bazel'));
+}
+
+/**
+ * Returns an array of all the files under a directory as relative
+ * paths to the directory.
+ */
+function listFiles(rootDir, subDir = '') {
+  const dir = path.posix.join(rootDir, subDir);
+  return fs
+      .readdirSync(dir)
+      // Filter out .bin folder since this is a folder created by yarn/npm that contains
+      // only symlinks and it breaks this function on Windows.
+      // For example, `node_modules/foo/.bin/foo` would be a symlink to `node_modules/.bin/foo`
+      .filter(f => !(f === '.bin' && fs.statSync(path.posix.join(dir, f)).isDirectory()))
+      // Also filter out test folders which contain symlinks in some packages
+      // such as `ecstatic/test/public/containsSymlink/more-problematic`
+      .filter(f => !(f === 'test' && fs.statSync(path.posix.join(dir, f)).isDirectory()))
```

Comment:
I'm sure there's some npm package where files we actually need at runtime are in a directory called `test`
If the problem is symlinks under windows, could we specifically filter out those symlinks somehow instead of this whole directory?

Reply:
It's a very odd issue. It ended up showing itself in OSX as well for the `test` folder but not the `.bin` folder.

Reply:
The failure is odd. It fails within `fs.readDirSync()`